File: python/perfect-numbers/perfect_numbers.py
def classify(number):
    """ A perfect number equals the sum of its positive divisors.

    :param number: int a positive integer
    :return: str the classification of the input integer
    """

    if number <= 0:
        raise ValueError('Classification is only possible for positive integers.')
    sum = 0
    for value in range(1, number // 2 + 1):
        if number % value == 0:
            sum += value
    return 'perfect' if sum == number else 'abundant' if sum > number else 'deficient'

File: clojure/perfect-numbers/src/perfect_numbers.clj
(ns perfect-numbers)

(defn factor-sum [number]
      (reduce + (filter #(zero? (rem number %)) (range 1 number)))
)

(defn classify [number]
      (if (< number 1) 
        (throw (IllegalArgumentException. "number cannot be negative")) 
        (let [sum (factor-sum number)] 
          (cond (> sum number) :abundant
            (< sum number) :deficient
            :else :perfect)))
)

